packages/components/src/site-thumbnail/index.tsx

```diff
@@ -0,0 +1,56 @@
+import classnames from 'classnames';
+import { ReactNode } from 'react';
+import './style.scss';
+import * as Mshots from './use-mshots';
+import { getTextColorFromBackground } from './utils';
+declare let require: ( path: string ) => typeof Mshots;
+
+const MSHOTS_OPTION = {
+	vpw: 1200,
+	vph: 1200,
+	w: 300,
+};
+
+type Props = {
+	backgroundColor?: string;
+	mShotsUrl?: string;
+	size?: string;
+	children?: ReactNode;
+};
+
+export const SiteThumbnail = ( {
+	backgroundColor,
+	children,
+	mShotsUrl,
+	size = 'small',
+}: Props ) => {
+	let useMshots: typeof Mshots;
+	let maybeImage: HTMLImageElement | undefined;
+
+	if ( mShotsUrl ) {
+		useMshots = require( './use-mshots' );
```

Comment:
This seems over complicated at the moment but haven't figured how to call the `usemShots` hook conditionally. We want to execute only if mShotsUrl is provided.

Reply:
[It's a bad idea to conditionally call hooks](https://reactjs.org/docs/hooks-rules.html).

We should bail out of the effect defined in `useMshotsImg`  if `mShotsUrl` is empty.

Reply:
The linter shows was a bad idea but was wondering if there was another way to do it without making the input to useMshotsImg optional, but it make sense to return early from there.